### PR TITLE
Update segmentation functionality

### PR DESCRIPTION
**Pull request by constantinpape**

### CODE REVIEW

File: scripts/cooper/run_mitochondria_segmentation.py [hunk 1]
```diff
@@ -7,8 +7,13 @@
 
 def run_mitochondria_segmentation(args):
     tiling = parse_tiling(args.tile_shape, args.halo)
-    segmentation_function = partial(segment_mitochondria, model_path=args.model_path, verbose=False, tiling=tiling)
-    inference_helper(args.input_path, args.output_path, segmentation_function, force=args.force, data_ext=args.data_ext)
+    segmentation_function = partial(
+        segment_mitochondria, model_path=args.model_path, verbose=False, tiling=tiling, scale=[0.5, 0.5, 0.5]
```
Comment (constantinpape) [inline scripts/cooper/run_mitochondria_segmentation.py:11]: I have hard-coded the scale factor to 1/2 here. Given how much better the mito segmentation looks with this we can stick with it from now on and don't need to experiment further.

File: scripts/cooper/run_mitochondria_segmentation.py [hunk 2]
```diff
@@ -28,6 +33,10 @@ def main():
         "--force", action="store_true",
         help="Whether to over-write already present segmentation results."
     )
+    parser.add_argument(
+        "--mask_path", help="The filepath to a tif file with a mask that will be used to restrict the segmentation."
+        "Can also be a directory with tifs if the filestructure matches input_path."
+    )
```
Comment (constantinpape) [inline scripts/cooper/run_mitochondria_segmentation.py:39]: The scripts now support passing a mask. The segmentation will be restricted to the inside of it (e.g. to just segment mitos / vesicles in one synaptic compartment)

File: synaptic_reconstruction/imod/export.py [hunk 3]
```diff
@@ -96,6 +135,9 @@ def export_segmentation(imod_path, mrc_path, object_id=None, output_path=None, r
         id_str = "" if object_id is None else f"for object {object_id}"
         raise RuntimeError(f"Segmentation extracted from {imod_path} {id_str} is empty.")
 
+    if depth_interpolation is not None:
+        segmentation = maximum_filter1d(segmentation, size=depth_interpolation, axis=0, mode="constant")
```
Comment (constantinpape) [inline synaptic_reconstruction/imod/export.py:139]: This works very well to close the gaps in the masks exported from IMOD, at least for compartment masks. `depth_interpolation=10` seems to be a good setting since they tend to annotate every 10th slice.

File: synaptic_reconstruction/inference/actin.py [hunk 4]
```diff
@@ -13,19 +14,45 @@
 # fragments and then binarize again.
 def segment_actin(
     input_volume: np.ndarray,
-    model_path: str,
+    model_path: Optional[str] = None,
+    model: Optional[torch.nn.Module] = None,
```
Comment (constantinpape) [inline synaptic_reconstruction/inference/actin.py:18]: All segmentation functions now take either a model path or an already loaded model.

File: synaptic_reconstruction/inference/actin.py [hunk 5]
```diff
@@ -13,19 +14,45 @@
...
 
-    pred = get_prediction(input_volume, model_path, tiling, verbose)
+    Returns:
+        The segmentation mask as a numpy array, or a tuple containing the segmentation mask
+        and the predictions if return_predictions is True.
+    """
+    if verbose:
+        print("Segmenting actin in volume of shape", input_volume.shape)
+    # Create the scaler to handle prediction with a different scaling factor.
+    scaler = _Scaler(scale, verbose)
```
Comment (constantinpape) [inline synaptic_reconstruction/inference/actin.py:49]: This class simplifies the scaling / rescaling of inputs / outputs.

File: synaptic_reconstruction/inference/postprocessing/vesicles.py [hunk 6]
```diff
@@ -22,17 +22,30 @@ def close_holes(vesicle_segmentation, closing_iterations=4, min_size=0, verbose=
     return closed_segmentation
 
 
-def filter_zborder_objects(segmentation):
-    """Filter any object that touches the upper or lower z-border.
+def filter_border_objects(segmentation: np.ndarray, z_border_only: bool = False) -> np.ndarray:
```
Comment (constantinpape) [inline synaptic_reconstruction/inference/postprocessing/vesicles.py:25]: This function now filters objects from all borders (unless `z_border_only=True`, then it only filters in the depth axis)

**Review by constantinpape — COMMENTED** [review]
@lufre1 @SarahMuth : here are all the changes to the inference functionality for the new vesicle and mito segmentation. I highlighted the important changes.
